### PR TITLE
Add bfloat16 value type support for CPU

### LINKED ISSUE #266: bfloat16 value type support
TFRA supports many value types including float16, but bfloat16 type is not there. Are there any specific reasons?

I tried to add a BF16 value type kernel for cuckoo hashtable op and it seems to work. And it is easier to converge than FP16 in training.

### PR DESCRIPTION
**Pull request by xiangzez**
# Description

Fixes #266

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [x] New Feature

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/recommenders-addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running yapf
    - [x] By running clang-format
- [x] This PR addresses an already submitted issue for TensorFlow Recommenders-Addons
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
tensorflow_recommenders_addons/dynamic_embedding/python/kernel_tests/dynamic_embedding_variable_test.py

### CODE REVIEW

**Comment by xiangzez** [comment]
I can't test mac platform so I leave it unsupported.  I'm also not sure if it is needed to add bf16 type to `redis_table_op` too.

File: tensorflow_recommenders_addons/dynamic_embedding/core/kernels/lookup_impl/lookup_table_op_cpu_impl_int64_bfloat16.cc [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
```
Comment (rhdong) [inline tensorflow_recommenders_addons/dynamic_embedding/core/kernels/lookup_impl/lookup_table_op_cpu_impl_int64_bfloat16.cc:1]: 2021->2022

Comment (xiangzez) [reply tensorflow_recommenders_addons/dynamic_embedding/core/kernels/lookup_impl/lookup_table_op_cpu_impl_int64_bfloat16.cc]: Done.

**Review by rhdong — APPROVED** [review]
LGTM

**Pull request merged**